pkg/providers/az-client.go

```diff
@@ -43,3 +46,34 @@ func (g *GraphServiceClient) GetApplicationObjectId(ctx context.Context, appId s
 	}
 	return *appObjectId, nil
 }
+
+type RoleAssignmentClient struct {
+	Client *armauthorization.RoleAssignmentsClient
+}
+
+type RoleAssignClient interface {
+	CreateRoleAssignment(ctx context.Context, objectId, roleId, scope, raUid string) error
+}
+
+var _ RoleAssignClient = &RoleAssignmentClient{}
+
+func (r *RoleAssignmentClient) CreateRoleAssignment(ctx context.Context, objectId, roleId, scope, raUid string) error {
```

Comment:
we could just directly make the interface have the `.CreateByID(ctx context.Context, roleAssignmentID string, parameters RoleAssignmentCreateParameters, options *RoleAssignmentsClientCreateByIDOptions) (RoleAssignmentsClientCreateByIDResponse, error)` method that way the mocks are mocking the Azure client instead of mocking our own method on a struct we made.

if we make the interface satisfiable by the existing arm client, we have a much stronger guarantee of type compatibility and testing coverage

then we can update
`var _ RoleAssignClient = &RoleAssignmentClient{}`<== our struct type
 to `var _ RoleAssignClient = &RoleAssignmentsClient{}` <== the Azure SDK's client type

pointing to https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v2#RoleAssignmentsClient

github link https://github.com/Azure/azure-sdk-for-go/blob/760c460eb365f6a95e8002062e0f2aceb3e267bc/sdk/resourcemanager/authorization/armauthorization/roleassignments_client.go#L25

the interface should be satisfied by this method https://github.com/Azure/azure-sdk-for-go/blob/760c460eb365f6a95e8002062e0f2aceb3e267bc/sdk/resourcemanager/authorization/armauthorization/roleassignments_client.go#L117